views/forms/SignInForm/SignInForm.test.jsx

```diff
@@ -1,24 +1,52 @@
 import React from 'react'
-import { render, screen } from '@testing-library/react'
-import LoginForm from '.'
-import FormProviderFixture from '../../../test/FormProviderFixture'
+import { render, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
 
-describe('Login Form', () => {
-  beforeEach(() => {
-    render(<LoginForm />, { wrapper: FormProviderFixture })
-  })
+import SignInForm from '.'
+
+describe('Sign In Form', () => {
+  const defaultProps = {
+    initialValues: {
+      username: '',
+      password: '',
+    },
+    onSubmit: () => {},
+  }
+  const elements = {
+    usernameField: () => screen.getByRole('textbox', { name: 'Username' }),
+    passwordField: () => screen.getByTestId('pw'),
+    submitBtn: () => screen.getByText('Sign in'),
+  }
+  const renderForm = (props = defaultProps) => {
+    render(<SignInForm {...props} />)
+  }
 
-  test('renders the text inputs', () => {
-    const usernameField = screen.getByLabelText('username')
-    const passwordField = screen.getByLabelText('password')
+  test('renders the text inputs and submit button', () => {
+    renderForm()
 
-    expect(usernameField).toBeInTheDocument()
-    expect(passwordField).toBeInTheDocument()
+    expect(elements.usernameField()).toBeInTheDocument()
+    expect(elements.passwordField()).toBeInTheDocument()
+    expect(elements.submitBtn()).toBeInTheDocument()
   })
 
-  test('renders the Sign In button', () => {
-    const signInButton = screen.getByText('Sign in')
+  test('calls the onSubmit prop when the form is submitted', async () => {
+    const user = userEvent.setup()
+    const onSubmit = jest.fn()
+    const props = { ...defaultProps, onSubmit }
+
+    renderForm(props)
+    await user.type(elements.usernameField(), 'myUsername')
```

Comment:
```suggestion
    renderForm(props)
    
    await user.type(elements.usernameField(), 'myUsername')
```

Reply:
these are grouped together so that the test examples follow the convention of

```
Setup

Execution

Assertions
```

Reply:
It helps for readability.